### ISSUE TITLE
provisioning not working with puppet

### ISSUE THREAD

**Issue by rtyler** [rtyler/vagrant-plugin#8]
```
ERROR: Shared folders that Puppet requires are missing on the virtual machine.
This is usually due to configuration changing after already booting the
machine. The fix is to run a `vagrant reload` so that the proper shared
folders will prepared and mounted on the VM. (Vagrant::Provisioners::PuppetError)
/home/tyler/.rvm/gems/jruby-1.6.5.1@vagrant-plugin/gems/vagrant-1.0.0/lib/vagrant/provisioners/puppet.rb:168:in `verify_shared_folders'
org/jruby/RubyArray.java:1612:in `each'
/home/tyler/.rvm/gems/jruby-1.6.5.1@vagrant-plugin/gems/vagrant-1.0.0/lib/vagrant/provisioners/puppet.rb:165:in `verify_shared_folders'
/home/tyler/.rvm/gems/jruby-1.6.5.1@vagrant-plugin/gems/vagrant-1.0.0/lib/vagrant/provisioners/puppet.rb:98:in `provision!'
/home/tyler/.rvm/gems/jruby-1.6.5.1@vagrant-plugin/gems/vagrant-1.0.0/lib/vagrant/action/vm/provision.rb:36:in `call'
org/jruby/RubyArray.java:1612:in `each'
/home/tyler/.rvm/gems/jruby-1.6.5.1@vagrant-plugin/gems/vagrant-1.0.0/lib/vagrant/action/vm/provision.rb:33:in `call'
/home/tyler/.rvm/gems/jruby-1.6.5.1@vagrant-plugin/gems/vagrant-1.0.0/lib/vagrant/action/warden.rb:33:in `call'
/home/tyler/.rvm/gems/jruby-1.6.5.1@vagrant-plugin/gems/vagrant-1.0.0/lib/vagrant/action/vm/check_accessible.rb:18:in `call'
/home/tyler/.rvm/gems/jruby-1.6.5.1@vagrant-plugin/gems/vagrant-1.0.0/lib/vagrant/action/warden.rb:33:in `call'
/home/tyler/.rvm/gems/jruby-1.6.5.1@vagrant-plugin/gems/vagrant-1.0.0/lib/vagrant/action/general/validate.rb:13:in `call'
/home/tyler/.rvm/gems/jruby-1.6.5.1@vagrant-plugin/gems/vagrant-1.0.0/lib/vagrant/action/warden.rb:33:in `call'
/home/tyler/.rvm/gems/jruby-1.6.5.1@vagrant-plugin/gems/vagrant-1.0.0/lib/vagrant/action/general/check_virtualbox.rb:23:in `call'
/home/tyler/.rvm/gems/jruby-1.6.5.1@vagrant-plugin/gems/vagrant-1.0.0/lib/vagrant/action/warden.rb:33:in `call'
/home/tyler/.rvm/gems/jruby-1.6.5.1@vagrant-plugin/gems/vagrant-1.0.0/lib/vagrant/action/builder.rb:92:in `call'
/home/tyler/.rvm/gems/jruby-1.6.5.1@vagrant-plugin/gems/vagrant-1.0.0/lib/vagrant/action/runner.rb:49:in `run'
/home/tyler/.rvm/gems/jruby-1.6.5.1@vagrant-plugin/gems/vagrant-1.0.0/lib/vagrant/util/busy.rb:19:in `busy'
/home/tyler/.rvm/gems/jruby-1.6.5.1@vagrant-plugin/gems/vagrant-1.0.0/lib/vagrant/action/runner.rb:49:in `run'
/home/tyler/.rvm/gems/jruby-1.6.5.1@vagrant-plugin/gems/vagrant-1.0.0/lib/vagrant/vm.rb:192:in `run_action'
/home/tyler/.rvm/gems/jruby-1.6.5.1@vagrant-plugin/gems/vagrant-1.0.0/lib/vagrant/vm.rb:164:in `provision'
/home/tyler/.rvm/gems/jruby-1.6.5.1@vagrant-plugin/gems/vagrant-1.0.0/lib/vagrant/command/provision.rb:23:in `execute'
/home/tyler/.rvm/gems/jruby-1.6.5.1@vagrant-plugin/gems/vagrant-1.0.0/lib/vagrant/command/base.rb:106:in `with_target_vms'
org/jruby/RubyArray.java:1612:in `each'
/home/tyler/.rvm/gems/jruby-1.6.5.1@vagrant-plugin/gems/vagrant-1.0.0/lib/vagrant/command/base.rb:101:in `with_target_vms'
/home/tyler/.rvm/gems/jruby-1.6.5.1@vagrant-plugin/gems/vagrant-1.0.0/lib/vagrant/command/provision.rb:19:in `execute'
/home/tyler/.rvm/gems/jruby-1.6.5.1@vagrant-plugin/gems/vagrant-1.0.0/lib/vagrant/cli.rb:38:in `execute'
/home/tyler/.rvm/gems/jruby-1.6.5.1@vagrant-plugin/gems/vagrant-1.0.0/lib/vagrant/environment.rb:163:in `cli'
/home/tyler/source/github/vagrant-plugin/models/vagrant_builder.rb:95:in `perform'
/home/tyler/.rvm/gems/jruby-1.6.5.1@vagrant-plugin/gems/jenkins-plugin-runtime-0.1.26/lib/jenkins/plugin/proxies/build_step.rb:14:in `perform'
/home/tyler/.rvm/gems/jruby-1.6.5.1@vagrant-plugin/gems/jenkins-plugin-runtime-0.1.26/lib/jenkins/plugin/proxies/build_step.rb:22:in `boolean_result'
/home/tyler/.rvm/gems/jruby-1.6.5.1@vagrant-plugin/gems/jenkins-plugin-runtime-0.1.26/lib/jenkins/plugin/proxies/build_step.rb:13:in `perform'

Build step 'Provision the Vagrant machine' marked build as failure
Build finished, destroying the Vagrant box
Finished: FAILURE
```

![wtf](http://25.media.tumblr.com/tumblr_m0lwgmyRRh1r4zlnbo1_500.gif)

The `/vagrant/` folder seems to be on the machine, I wonder if something else gets magically mapped that I'm not aware of.

**Comment by rtyler** [comment]
Crap, just tested this outside of the plugin with regular Vagrant. It looks like the `vagrant provision` command is broken in 1.0.1

**Comment by rtyler** [comment]
This will be resolved in the Vagrant gem with [pull request #803](https://github.com/mitchellh/vagrant/pull/803) but until then, we'll just bundle a patched version of Vagrant :/
